desc/objectives/_coils.py

```diff
@@ -597,3 +598,274 @@ def compute(self, params, constants=None):
         data = tree_flatten(data, is_leaf=lambda x: isinstance(x, dict))[0]
         out = jnp.concatenate([dat["torsion"] for dat in data])
         return out
+
+
+class ToroidalFlux(_Objective):
+    """Target the toroidal flux in an equilibrium from a magnetic field.
+
+    Parameters
+    ----------
+    eq : Equilibrium, optional
+        Equilibrium that will be optimized to satisfy the Objective.
+    field : MagneticField
+        MagneticField object, the parameters of this will be optimized
+        to minimize the objective.
+    target : {float, ndarray}, optional
+        Target value(s) of the objective. Only used if bounds is None.
+        Defaults to eq.Psi. Must be broadcastable to Objective.dim_f.
+    bounds : tuple of {float, ndarray}, optional
+        Lower and upper bounds on the objective. Overrides target.
+        Both bounds must be broadcastable to to Objective.dim_f
+    weight : {float, ndarray}, optional
+        Weighting to apply to the Objective, relative to other Objectives.
+        Must be broadcastable to to Objective.dim_f
+    normalize : bool, optional
+        Whether to compute the error in physical units or non-dimensionalize.
+        Note: has no effect on this objective.
+        FIXME: add normalization for the B part of this objective
+    normalize_target : bool
+        Whether target should be normalized before comparing to computed values.
+        if `normalize` is `True` and the target is in physical units, this should also
+        be set to True.
+        Note: has no effect on this objective.
+    loss_function : {None, 'mean', 'min', 'max'}, optional
+        Loss function to apply to the objective values once computed. This function
+        is called on the raw compute value, before any shifting, scaling, or
+        normalization. Note: has no effect for this objective
+    deriv_mode : {"auto", "fwd", "rev"}
+        Specify how to compute jacobian matrix, either forward mode or reverse mode AD.
+        "auto" selects forward or reverse mode based on the size of the input and output
+        of the objective. Has no effect on self.grad or self.hess which always use
+        reverse mode and forward over reverse mode respectively.
+    source_grid : Grid, optional
+        Collocation grid containing the nodes to evaluate field source at on
+        the winding surface. (used if e.g. field is a CoilSet or
+        FourierCurrentPotentialField). Defaults to the default for the
+        given field, see the docstring of the field object for the specific default.
+    eval_grid : Grid, optional
+        Collocation grid containing the nodes to evaluate the normal magnetic field at
+        plasma geometry at. Defaults to a LinearGrid(L=eq.L_grid, M=eq.M_grid,
+        zeta=jnp.array(0.0), NFP=eq.NFP).
+    external_field : MagneticField, optional
```

Comment:
rename to `field`

Reply:
and shouldn't be optional

Reply:
This was to allow the inclusion of a field that was not being optimized like a background TF field. But with the optimizable collection stuff this should not be needed anymore, we can just pass FixParameter to a SumMagneticField